netty/src/main/java/io/grpc/netty/NettyServerTransport.java

```diff
@@ -59,6 +63,7 @@
   private final ChannelPromise channelUnused;
   private final ProtocolNegotiator protocolNegotiator;
   private final int maxStreams;
+  private NettyServerHandler grpcHandler;   // Only set once during start()
```

Comment:
this will still be racy if there isnt a HB between assigning and accessing.   start() does not HB event loop access

Reply:
turned field into volatile